sql/sql_utils.cc

```diff
@@ -38,6 +43,65 @@ std::string escapeSql(const char * str)
     return escapeSql(std::string(str));
 }
 
+bool matchSqlFilter(const Utf8String& valueString, const Utf8String& filterString)
+{
+    if (filterString.empty() || valueString.empty())
+        return false;
+
+    Utf8String regExFilter;
+
+    for (const auto& filterChar : filterString) {
```

Comment:
If I understand correctly, the purpose here is to transform a like expression into a regex.  I believe that this does not handle the escape mechanism in like expression.  If I read the doc correctly, ```[]``` are used as escape.  Here we transform them into character to match.  Also, there are several special character in regex (e.g. $, ^) that are not special in like expression that would need to be escaped as well.